kalite/distributed/management/commands/retrievecontentpack.py

```diff
@@ -140,3 +143,54 @@ def extract_content_db(zf, lang):
     with open(content_db_path, "wb") as f:
         dbfobj = zf.open("content.db")
         shutil.copyfileobj(dbfobj, f)
+
+
+def extract_subtitles(zf, lang):
+    SUBTITLE_DEST_DIR = get_subtitle_path(lang_code=lang)
+    SUBTITLE_ZIP_DIR = "subtitles/"
+
+    ensure_dir(SUBTITLE_DEST_DIR)
+
+    subtitles = (s for s in zf.namelist() if SUBTITLE_ZIP_DIR in s)
+
+    for subtitle in subtitles:
+        # files inside zipfiles may come with leading directories in their
+        # names, like subtitles/hotdog.vtt. We'll only want the actual filename
+        # (hotdog.vtt) when extracting as that's what KA Lite expects.
+
+        subtitle_filename = os.path.basename(subtitle)
+        subtitle_dest_path = os.path.join(SUBTITLE_DEST_DIR, subtitle_filename)
+
+        subtitle_fileobj = zf.open(subtitle)
+
+        with open(subtitle_dest_path, "w") as dest_fileobj:
+            shutil.copyfileobj(subtitle_fileobj, dest_fileobj)
+
+
+def extract_assessment_items(zf, lang):
+    assessment_zip_dir = "assessment_resources/"
+
+    channel = "khan"
+
+    assessment_dest_dir = os.path.join(content_settings.ASSESSMENT_ITEM_ROOT, channel)
+
+    ensure_dir(assessment_dest_dir)
+
+    items = (s for s in zf.namelist() if assessment_zip_dir in s)
+
+    for item in items:
+        # files inside zipfiles may come with leading directories in their
+        # names, like subtitles/hotdog.vtt. We'll only want the actual filename
+        # (hotdog.vtt) when extracting as that's what KA Lite expects.
+
+        filename = os.path.basename(item)
+        subfolder = filename[:3]
+        assessment_subfolder = os.path.join(assessment_dest_dir, subfolder)
+
+        ensure_dir(assessment_subfolder)
+
+        zip_item_fileobj = zf.open(item)
+
+        assessment_dest_path = os.path.join(assessment_subfolder, filename)
+        with open(assessment_dest_path, "w") as item_fileobj:
+            shutil.copyfileobj(zip_item_fileobj, item_fileobj)
```

Comment:
I think this and the other use of `shutil.copyfileobj` might be really slow... now benchmarking.